rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -72,17 +75,48 @@ def import_rmw_implementation():
 
     If :py:func:`select_rmw_implementation` has not previously been called then
     a default implementation will be selected implicitly before loading.
+
+    :raises: :py:class:`NoImplementationAvailableException` if there are no
+        implementations available.
     """
     global __rmw_implementation_module
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
+    available_implementations = get_rmw_implementations()
     if __selected_rmw_implementation is None:
         logger = logging.getLogger('rclpy')
-        select_rmw_implementation(get_rmw_implementations()[0])  # select the first one
+        if available_implementations:
+            select_rmw_implementation(available_implementations[0])  # select the first one
+        else:
+            raise NoImplementationAvailableException()
         logger.debug("Implicitly selecting the '{0}' rmw implementation."
                      .format(__selected_rmw_implementation))
     module_name = '._rclpy__{rmw_implementation}'.format(
         rmw_implementation=__selected_rmw_implementation,
     )
-    __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
+    try:
+        __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
+    except ImportError as exc:
+        if "No module named 'rclpy._rclpy__" in str(exc):
```

Comment:
I wonder if you could do slightly better than hardcoding the module prefix here. If the module name changed from `rclpy._rclpy__rmw_....` to something else then this logic would break. Wouldn't something like `"No module named 'rclpy.{0}".format(module_name)` work?